src/db/pool_unprepared_statement.cr

```diff
@@ -0,0 +1,21 @@
+module DB
+  # Represents a statement to be executed in any of the connections
+  # of the pool. The statement is not be executed in a non prepared fashion.
+  # The execution of the statement is retried according to the pool configuration.
+  #
+  # See `PoolStatement`
+  class PoolUnpreparedStatement < PoolStatement
+    def initialize(db : Database, query : String)
+      super
+    end
+
+    protected def do_close
+      # unprepared statements do not need to be release in each connection
+    end
+
+    # builds a statement over a real connection
+    private def build_statement
+      @db.pool.checkout.unprepared.build(@query)
```

Comment:
since it is a struct I would bet llvm will inline that. We could even add the @[AlwaysInline] maybe. I prefer use public api if possible. build_(un)prepared_statements are to be defined by drivers but I wouldn't suggest to use them as public api: they are too verbose.